sklearn/conftest.py

```diff
@@ -227,3 +229,76 @@ def pytest_configure(config):
         matplotlib.use("agg")
     except ImportError:
         pass
+
+
+# Definition of the random_seed fixture. See the docstring of the fixture for
+# more information.
+
+RANDOM_SEED_RANGE = list(range(100))  # All seeds in [0, 99] should be valid.
+random_seed_var = environ.get("SKLEARN_TESTS_RANDOM_SEED")
+if random_seed_var is None:
+    # If the environment variable is not defined, pick-up one seed at random in
+    # the range of admissible random seeds. Note, to make sure that all
+    # pytest-xdist workers see the same seed, we seed the meta-random number
+    # generator with a value derived from the year and the day.
+    rng = Random(int(datetime.now().strftime("%Y%j")))
+    random_seeds = [rng.choice(RANDOM_SEED_RANGE)]
+elif random_seed_var == "all":
+    random_seeds = RANDOM_SEED_RANGE
+else:
+    if "-" in random_seed_var:
+        start, stop = random_seed_var.split("-")
+        random_seeds = list(range(int(start.strip()), int(stop.strip()) + 1))
+    else:
+        random_seeds = [int(s.strip()) for s in random_seed_var.split(",")]
+
+if min(random_seeds) < 0 or max(random_seeds) > 99:
+    raise ValueError(
+        "The value(s) of the environment variable SKLEARN_TESTS_RANDOM_SEED "
+        f"must be in the range [0, 99] (or 'all'), got: {random_seed_var}"
+    )
+
+
+def pytest_report_header(config):
+    if random_seeds == RANDOM_SEED_RANGE:
+        seed_values = "all"
+    else:
+        seed_values = ",".join(str(s) for s in random_seeds)
+    return (
+        "To reproduce this test run, set the following environment variable:\n"
+        f'    SKLEARN_TESTS_RANDOM_SEED="{seed_values}"'
+    )
+
+
+@pytest.fixture(params=random_seeds)
+def random_seed(request):
+    """Fixture to ask for a random yet controllable random seed.
+
+    All tests that use this fixture accept the contract that they should
+    deterministically pass for any seed value from 0 to 99 included.
+
+    If the SKLEARN_TESTS_RANDOM_SEED environment variable is not set (which
+    should be the default, in particular on the CI), the fixture will randomly
+    choose on seed in the above range and all fixtured tests will run for that
+    specific seed. This ensures that over time, our CI will run all tests with
+    different seeds while keeping the test duration of a single run of the full
+    test suite limited. This will enforce that the tests assertions of tests
+    written to use this fixture are not dependent on a specific seed value.
+
+    The range of admissible seed values is limited to [0, 99] because it is
+    often not possible to write a test that can work for any possible seed.
```

Comment:
I like this idea of chaos testing.

```suggestion
    If the SKLEARN_TESTS_RANDOM_SEED environment variable is not set (which
    should be the default, in particular on the CI), the fixture will randomly
    choose one seed in the above range and all fixtured tests will run for that
    specific seed. This ensures that over time, our CI will run all tests with
    different seeds while keeping the test duration of a single run of the full
    test suite limited. This will enforce that the tests assertions of tests
    written to use this fixture are not dependent on a specific seed value.

    The range of admissible seed values is limited to [0, 99] because it is
    often not possible to write a test that can work for any possible seed.
```

Reply:
I changed the phrasing concurrently to your suggestion and the typo went away.